src/main/java/com/growup/todoary/service/dto/response/CategoryResponse.java

```diff
@@ -0,0 +1,19 @@
+package com.growup.todoary.service.dto.response;
+
+import com.growup.todoary.domain.Category;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+
+@Builder
+@AllArgsConstructor
```

Comment:
Builder를 사용하는데 AllargsConstructor도 열려있다면 다른 개발자가 사용할 때 두 방식을 혼용해서 사용할 수도 있겠네요~
개인적으로 Builder 혹은 생성자 방식 둘중 하나만 사용하는것을 좋아하는데 어떻게 생각하시나요?

+) Builder 방식만 사용하도록 지정하고싶다면 다음 옵션을 추가할 수 있습니다.
```suggestion
@AllArgsConstructor(access = AccessLevel.PRIVATE)
```

Reply:
Builder만 있을 때는 `error: constructor CategoryResponse in class CategoryResponse cannot be applied to given types; ` 에러가 났습니다. 찾아보니 모든 생성자를 받는 함수가 없어서 그렇다고 하여 @AllargsConstructor를 붙였습니다..!

(access = AccessLevel.PRIVATE)를 붙이는게 좋을 것 같습니다. 접근제어자 private과 같은 의미이고, Builder는 내부에 있는 거라 접근 가능한 것으로 이해 했는데 맞나요?

Reply:
네 맞아요~
Builder는 내부에있는거라기보단 접근제어자가 public이기 때문에 접근 가능하다고 말하는게 맞겠네요~

Lombok은 개발자가 보기 쉽게 코드를 줄여주는 역할을 합니다.
실제로 컴파일 시에는 다음과 같은 코드가 생기는거죠~

```java
// @AllArgsConstructor는 아래 코드로 변환된다.

public CategoryResponse(Long id, String name) {
    this.id = id;
    this.name = name;
}

// 같은 의미로 @AllArgsConstructor(access = AccessLevel.PRIVATE)는 다음과 같이 생성된다.

private CategoryResponse(Long id, String name) {
    this.id = id;
    this.name = name;
}

// @Builder는 실제로 다음과 같은 코드를 생성한다.

public static class TodoResponseBuilder {
    private Long id;
    private String name;
    private LocalDate date;

    TodoResponseBuilder() {
    }

    public TodoResponseBuilder id(final Long id) {
        this.id = id;
        return this;
    }

    public TodoResponseBuilder name(final String name) {
        this.name = name;
        return this;
    }

    public TodoResponseBuilder date(final LocalDate date) {
        this.date = date;
        return this;
    }

    public TodoResponse build() {
        return new TodoResponse(this.id, this.name, this.date);
    }

    public String toString() {
        return "TodoResponse.TodoResponseBuilder(id=" + this.id + ", name=" + this.name + ", date=" + this.date + ")";
    }
}
```